src/ir/from_ast/astconv.rs

```diff
@@ -57,6 +56,18 @@ impl<'prog> BuildCtx<'prog> {
                 name.pos(),
             )),
         };
+
+        // Extend the binding with the let-bound parameters in the signature
+        // Bindings can mention previous bindings so we add bindings as we
+        // go along.
+        comp.sig_binding.iter().for_each(
+            |ast::ParamBind { param, default }| {
+                let e = default.clone().unwrap().resolve(&binding);
+                binding.extend([(param.copy(), e)])
```

Comment:
I will probably see as I continue reading, but where do we exclude the sig-bound parameters from the instance's parameter list? Because if I understand this correctly, by extending the binding that we use to make the `ir::Instance`, there will be parameters there that can't be passed in (causing the errors we saw before)?

Reply:
So the `ir::Instance` object above takes the normal `binding` before we extend it to include the `sig_binding` stuff. This means that the instance gets parameter list we want while the bindings we store track the extra sig-bound parameters.